src/components/ChartJsWrapper/index.jsx

```diff
@@ -33,50 +34,70 @@ const ChartJsWrapper = ({
   type,
   chartHeight,
   spinnerSize,
-  missingDataError = false,
-}) => {
-  let showError;
+}) =>
+  (() => {
+    if (!data) {
+      return (
+        <div
```

Comment:
The problem of moving the rendering of the NO DATA case scenario up here is that there are now two different locations to determine what does this method render.
I would prefer that the two were together. Your call.

Reply:
I find this clearer.  When `!data` we show `<CircularProgress>`

The code changed to expose the fact it is actually has three distinct states, and shows three distinct compoenents:  https://github.com/mozilla-frontend-infra/firefox-health-dashboard/issues/351